src/charm.py

```diff
@@ -809,6 +811,11 @@ def _charm_logging_ca_cert(self) -> Optional[str]:
             return self._ca_cert_path
         return None
 
+    def _add_readiness_check(self, new_layer: Layer):
+        """Add readiness check to a pebble layer."""
+        new_layer.checks["ready"] = Check(
+            "ready", {"override": "replace", "http": {"url": "http://localhost:3100/ready"}}
```

Comment:
The readiness check, when pebble will be able to trigger an event on success, would be very useful, and in some cases could replace the logic we run on pebble-ready.

With our current TLS design, would need to use the external url when TLS is enabled.